apps/users/views.py

```diff
@@ -1,3 +1,152 @@
+        return Response(data, status.HTTP_201_CREATED)
+
+    @action(detail=False, methods=['POST'], url_path='send-verification', serializer_class=UserEmailSerializer)
+    def send_verification(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        user = get_object_or_404(User, email=serializer.initial_data['email'])
+
+        if user.is_active:
+            return Response({'user': 'The user with this email is verified'}, status.HTTP_403_FORBIDDEN)
+
+        user.generate_code(is_registration=True)
+        return Response()
+
+    @action(detail=False, methods=['POST'], serializer_class=UserVerificationSerializer)
+    def confirm(self, request, *args, **kwargs):  # TODO Create stripe ID
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        validated_data = serializer.validated_data
+
+        verification_code = get_object_or_404(UserVerification, code=validated_data['code'], is_completed=False)
+        user = verification_code.verify_user()
+
+        return Response(UserSerializer(user).data)
+
+
+class UserLoginResetViewSet(GenericViewSet):
+    """
+    Used for authentication and managing existing accounts.
+    """
+    serializer_class = UserSerializer
+    queryset = User.objects.all()
+    permission_classes = (AllowAny,)
+
+    @action(detail=False, methods=['POST'], serializer_class=UserEmailSerializer)
+    def send_reset_code(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        data = serializer.initial_data
+
+        user = get_object_or_404(User, email=data['email'], is_active=True)
+
+        user.generate_code(is_registration=False)
+
+        return Response()
+
+    @action(detail=False, methods=['POST'], serializer_class=UserLoginResetSerializer)
+    def reset(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        validated_data = serializer.data
+
+        verification_code = get_object_or_404(
+            UserVerification, code=validated_data['code'], is_completed=False, is_registration=False)
+
+        user = get_object_or_404(User, id=verification_code.user_id)
+
+        user.set_password(validated_data['password'])
+        verification_code.is_completed = True
+
+        verification_code.save()
+        user.save()
+
+        return Response(UserSerializer(user).data)
+
+
+class UserProfileViewSet(GenericViewSet, mixins.RetrieveModelMixin, mixins.UpdateModelMixin):
+    """
+    Used for authentication and managing existing accounts.
+    """
+    serializer_class = UserProfileSerializer
+    queryset = User.objects.all()
+    permission_classes = (IsAuthenticated, IsAdminOrItself | ReadOnly,)
+    parser_classes = (MultiPartParser, )
+
+    @action(detail=False, methods=['GET'])
+    def current(self, request, *args, **kwargs):
```

Comment:
in viewset de user profile de seteaza in get_queryset userul curent, si toate actiunile se fac pentru userul logat